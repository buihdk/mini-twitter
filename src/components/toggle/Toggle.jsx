import React, { memo } from 'react';
import PropTypes from 'prop-types';

import { noop } from '../../utils';
import './Toggle.scss';

const Toggle = ({ lLabel, rLabel, isRain, toggle }) => (
  <div className="toggleContainer">
    <span className="label">{lLabel}</span>
    <label className="switch" htmlFor="toggle">
      <input type="checkbox" id="toggle" onClick={() => toggle(!isRain)} />
      <span className="slider round" />
    </label>
    <span className="label">{rLabel}</span>
  </div>
);

Toggle.propTypes = {
  lLabel: PropTypes.string,
  rLabel: PropTypes.string,
  isRain: PropTypes.bool,
  toggle: PropTypes.func,
};

Toggle.defaultProps = {
  lLabel: '',
  rLabel: '',
  isRain: false,
  toggle: noop,
};

export default memo(Toggle);
