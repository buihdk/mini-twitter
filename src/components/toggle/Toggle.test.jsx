import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Toggle from './Toggle';

describe('Toggle', () => {
  test('renders without crashing', () => {
    const wrapper = shallow(<Toggle />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
