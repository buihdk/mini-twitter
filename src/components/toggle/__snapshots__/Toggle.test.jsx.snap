// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle renders without crashing 1`] = `
<div
  className="toggleContainer"
>
  <span
    className="label"
  />
  <label
    className="switch"
    htmlFor="toggle"
  >
    <input
      id="toggle"
      onClick={[Function]}
      type="checkbox"
    />
    <span
      className="slider round"
    />
  </label>
  <span
    className="label"
  />
</div>
`;
